atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/TimeLockServerTest.java

```diff
@@ -15,102 +15,52 @@
-public class TimeLockServerTest {
-    private static final String LOCAL_ADDRESS_STRING = "localhost:12345";
-    private static final Address LOCAL_ADDRESS = new Address(LOCAL_ADDRESS_STRING);
-    private static final TimeLockServerConfiguration TIMELOCK_CONFIG = new TimeLockServerConfiguration(
-            ImmutableAtomixConfiguration.builder()
-                    .storageLevel(StorageLevel.MEMORY)
-                    .build(),
-            ImmutableClusterConfiguration.builder()
-                    .localServer(LOCAL_ADDRESS_STRING)
-                    .addServers(LOCAL_ADDRESS_STRING)
-                    .build(),
-            ImmutableSet.of(String.format("%s:%s", LOCAL_ADDRESS.host(), LOCAL_ADDRESS.port())));
-
-    private final TimeLockServer server = new TimeLockServer();
-    private final Environment environment = mock(Environment.class);
-    private final List<LifeCycle.Listener> listeners = new ArrayList<>();
-
-    @Before
-    public void setupEnvironment() {
-        when(environment.jersey()).thenReturn(mock(JerseyEnvironment.class));
-
-        LifecycleEnvironment lifecycle = mock(LifecycleEnvironment.class);
-        when(environment.lifecycle()).thenReturn(lifecycle);
-        doAnswer(inv -> listeners.add((LifeCycle.Listener) inv.getArguments()[0]))
-                .when(lifecycle).addLifeCycleListener(any());
-    }
-
-    @After
-    public void sendShutdownToListeners() {
-        LifeCycle event = mock(LifeCycle.class);
-        listeners.forEach(listener -> listener.lifeCycleStopping(event));
-        listeners.forEach(listener -> listener.lifeCycleStopped(event));
+public class TimeLockServerTest extends ServerImplementationTest {
+    private static final String LOCAL_ADDRESS = "localhost:8080";
+    private static final String TEST_CLIENT = "test";
+
+    private ServerImplementation serverImplementation;
+
+    @Override
+    protected TimeLockServerConfiguration getConfiguration() {
+        TimeLockAlgorithmConfiguration algorithmConfiguration = mock(TimeLockAlgorithmConfiguration.class);
+        serverImplementation = mock(ServerImplementation.class);
+        when(algorithmConfiguration.createServerImpl()).thenReturn(serverImplementation);
+        when(serverImplementation.createInvalidatingTimeLockServices(any())).thenReturn(mock(TimeLockServices.class));
+
+        return new TimeLockServerConfiguration(
+                algorithmConfiguration,
+                ImmutableClusterConfiguration.builder()
+                        .localServer(LOCAL_ADDRESS)
+                        .addServers(LOCAL_ADDRESS)
+                        .build(),
+                ImmutableSet.of(TEST_CLIENT));
     }
 
-    @Test
-    public void atomixIsRunningAfterSuccessfulStartup() throws IOException {
-        server.run(TIMELOCK_CONFIG, environment);
-        tryToConnectToAtomixPort();
+    @Override
+    protected void verifyPostStartupSuccess() {
```

Comment:
Okay. As discussed offline, verifying a method was called on a non-mock instantiation of the `ServerImplementation` is messy. I've refactored this to a `TimeLockServerTest` which is not part of the abstract class hierarchy and renamed the method there to be more descriptive as suggested. I'll also clarify the method for the abstract tests to say `verifyStateAfterStartupSuccess` (emphasis on state).